core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -211,6 +211,15 @@ import akka.stream.scaladsl.Source
           result.flatMap(_ => deleteDeletedToSeqNr(persistenceId))
         else result.map(_ => Done)
       result2.pipeTo(sender())
+
+    case HealthCheckQuery =>
+      session
+        .selectOne(healthCheckStatement)
+        .map {
+          case Some(_) => HealthCheckResponse(true)
+          case None    => HealthCheckResponse(false)
```

Comment:
I wonder if it is a good conclusion that 0 rows mean failed health check.
Maybe we should treat empty result as a succeeded health check and only failed executions would mean that the cluster is not healthy.
I don't know what contract users would expect. What is your opinion?

Reply:
I think 0 rows can still be treated as `true`. It wasn't an error connecting to Cassandra. Some custom query might want to select from an empty table for some reason.

Reply:
I changed this behaviour. Done